src/thing/Transcriber.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include "thing/Transcriber.h"
+
+namespace thing {
+namespace {
+  const char* kSubPathDigitalOut = "/digital_out";
+  const char* kSubPathDigitalIn = "/digital_in";
+  const char* kSubPathAnalogOut = "/analog_out";
+  const char* kSubPathAnalogIn = "/analog_in";
+}  // namespace
+
+
+Transcriber::Transcriber(const Config& config) {
+  Init(config);
+}
+
+void Transcriber::UpdateConfig(const Config& config) {
+  Init(config);
+}
+
+void Transcriber::Init(const Config& config) {
+  path_ = config.path;
+  pin_digital_out_ = config.pin_digital_out;
+  pin_digital_in_ = config.pin_digital_in;
+  pin_analog_out_ = config.pin_analog_out;
+  pin_analog_in_ = config.pin_analog_in;
+
+  fbase_.reset(new Firebase(config.host, config.auth));
+  stream_ = fbase_->streamPtr(path_);
+}
+
+void Transcriber::SetValue(const std::string& path, const std::string& value) {
+  stream_.reset(nullptr);
```

Comment:
Yes, I added http->end() to the FirebaseCall destructor, I don't think it is necessary as the Esp8266HttpClient destructor is doing this too but since we have a portability layer everything in the future may not.